packages/gatsby-plugin-page-creator/src/__tests__/gatsby-node.js

```diff
@@ -18,6 +18,59 @@ describe(`JavaScript page creator`, () => {
     expect(files.filter(file => validatePath(file.path)).length).toEqual(1)
   })
 
+  it(`filters out files that start with dot`, () => {
+    const files = [
+      {
+        path: `.eslintrc`,
+      },
+      {
+        path: `something/.eslintrc`,
+      },
+      {
+        path: `something/.eslintrc.js`,
+      },
+      {
+        path: `something/blah.js`,
+      },
+      {
+        path: `.markdownlint.json`,
+      },
+      {
+        path: `something/.markdownlint.json`,
+      },
+    ]
+
+    expect(files.filter(file => validatePath(file.path)).length).toEqual(1)
+  })
+
+  it(`filters out json and yaml files`, () => {
+    const files = [
+      {
+        path: `something/blah.js`,
+      },
+      {
+        path: `something/otherConfig.yml`,
+      },
+      {
+        path: `config.yaml`,
+      },
+      {
+        path: `somefile.json`,
+      },
+      {
+        path: `somefile.js`,
+      },
+      {
+        path: `dir1/file.json`,
+      },
+      {
+        path: `dir1/dir2/file.json`,
+      },
+    ]
+
+    expect(files.filter(file => validatePath(file.path)).length).toEqual(2)
```

Comment:
Another way you could write this would be something like this

```js
const expected = [
  { path: `something/blah.js` },
  { path: `somefile.js` }
]

const files = expected.concat([]) // other files

expect(files.filter(file => validatePath(file.path)).length).toEqual(expected.length)
```

Reply:
hey @DSchau, I was just sticking with the format of the tests that were already there. If there is a new pattern for these types or tests, or you want me to update this and the other nit happy to do so.

Reply:
Should've looked! Up to you :) I'm all for improvements, and this is probably a good time to do them! What do you think?